test/test_linalg.py

```diff
@@ -2295,6 +2295,21 @@ def test_eig_errors_and_warnings(self, device, dtype):
             with self.assertRaisesRegex(RuntimeError, "tensors to be on the same device"):
                 torch.linalg.eig(a, out=(out_w, out_v))
 
+    @skipCPUIfNoLapack
+    @dtypes(*floating_and_complex_types())
+    def test_eig_with_nan(self, device, dtype):
```

Comment:
This test is fine as it is, but I think it can be rewritten to contain fewer lines with a for loop, something like:
```
for batch_dim in [(), (10,)]:
    a = make_tensor((*batch_dim, 5, 5), device=device, dtype=dtype)
    a[..., -1, -1] = float('nan')
    with self.assertRaisesRegex(...):
        torch.linalg.eig(a)
```

Reply:
There's no test for inputs containing infinity values.